src/Catalyst.Cli/Commands/CommandContext.cs

```diff
@@ -0,0 +1,133 @@
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Catalyst.Abstractions.Cli;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.IO.Transport;
+using Catalyst.Abstractions.Rpc;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Network;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.Rpc.Client;
+using Catalyst.Protocol.Peer;
+using Dawn;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+
+namespace Catalyst.Cli.Commands
+{
+    public class CommandContext : ICommandContext
+    {
+        private readonly IList<IRpcClientConfig> _rpcNodeConfigs;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// </summary>
+        public CommandContext(IConfigurationRoot config,
+            ILogger logger,
+            IUserOutput userOutput,
+            IRpcClientFactory rpcClientFactory,
+            ICertificateStore certificateStore,
+            ISocketClientRegistry<IRpcClient> socketClientRegistry)
+        {
+            _logger = logger;
+            _rpcNodeConfigs = RpcClientSettings.BuildRpcNodeSettingList(config);
+
+            SocketClientRegistry = socketClientRegistry;
+            PeerId = GetPeerIdentifierFromCliConfig(config);
+            RpcClientFactory = rpcClientFactory;
+            CertificateStore = certificateStore;
+            UserOutput = userOutput;
+        }
+
+        public PeerId PeerId { get; }
+
+        public IRpcClientFactory RpcClientFactory { get; }
+
+        public ICertificateStore CertificateStore { get; }
+
+        public IUserOutput UserOutput { get; }
+
+        public ISocketClientRegistry<IRpcClient> SocketClientRegistry { get; }
+
+        /// <inheritdoc cref="GetConnectedNode" />
+        public IRpcClient GetConnectedNode(string nodeId)
+        {
+            Guard.Argument(nodeId, nameof(nodeId)).NotNull().NotEmpty().Compatible<string>();
+            var nodeConfig = _rpcNodeConfigs.SingleOrDefault(node => node.NodeId.Equals(nodeId));
+            Guard.Argument(nodeConfig, nameof(nodeConfig)).NotNull();
+
+            var registryId = SocketClientRegistry.GenerateClientHashCode(
+                EndpointBuilder.BuildNewEndPoint(nodeConfig?.HostAddress, nodeConfig.Port));
+
+            var nodeRpcClient = SocketClientRegistry.GetClientFromRegistry(registryId);
+            Guard.Argument(nodeRpcClient).Require(IsSocketChannelActive(nodeRpcClient));
+
+            return nodeRpcClient;
+        }
+
+        /// <inheritdoc cref="GetNodeConfig" />
+        public IRpcClientConfig GetNodeConfig(string nodeId)
+        {
+            Guard.Argument(nodeId, nameof(nodeId)).NotNull().NotEmpty().Compatible<string>();
+
+            var nodeConfig = _rpcNodeConfigs.SingleOrDefault(config => config.NodeId.Equals(nodeId));
+
+            if (nodeConfig != null) return nodeConfig;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)